vampyr/operators.cpp

```diff
@@ -0,0 +1,188 @@
+             "GreensKernel: DerivativeKernel");
+
+    //DerivativeOperator
+    py::class_<DerivativeOperator<D>> derivativeoperator(m, "Derivative Operator");
+    derivativeoperator.def(py::init<MultiResolutionAnalysis<D>>());
+
+    //GreensKernel
+	  py::class_<GreensKernel> greensKernel(m, "GreensKernel");
+
+    //HelmholtzKernel
+    py::class_<HelmholtzKernel>(m, "HelmholtzKernel", greensKernel)
+        .def(py::init<double, double, double, double>(),
+             "m"_a,
+             "eps"_a,
+             "r_min"_a,
+             "r_max"_a,
+             "GreensKernel: HelmholtzKernel");    
+
+    //HelmholtzOperator
+    py::class_<HelmholtzOperator, ConvolutionOperator<D>>(m, "HelmholtzOperator", py::multiple_inheritance())
+        .def(py::init<MultiResolutionAnalysis<D> &, double, double>(),
+             "MRA"_a,
+             "mu"_a,
+             "precision"_a = -1.0,
+             "ConvolutionOperator: HelmholtzOperator exp(-mu*r)/|r-r'|");
+
+    //IdentityConvolution
+	  py::class_<IdentityConvolution<D>> identityconvop(m, "IdentityConvolution");
+    identityconvop.def(py::init<MultiResolutionAnalysis<D> &, double>());
+
+    //IdentityKernel
+    py::class_<IdentityKernel>(m, "IdentityKernel", greensKernel)
+        .def(py::init<double>(),
+             "eps"_a,
+             "GreensKernel: IdentityKernel");
+
+    //MWOperator
+    py::class_<MWOperator> mwoperator(m, "MWOperator");
+    mwoperator.def(py::init<MultiResolutionAnalysis<2>>())
+        .def("size", &MWOperator::size, "Size")
+        .def("push_back", &MWOperator::push_back, "Push")
+        .def("clear", &MWOperator::clear, "Clear")
+        .def("getMaxBandWidth", &MWOperator::getMaxBandWidth, "Get")
+        .def("calcBandWidths", &MWOperator::calcBandWidths, "Calc")
+        .def("clearBandWidths", &MWOperator::clearBandWidths, "Clear");
+
+    //OperatorState
+    py::class_<OperatorState<D>> operatorstate(m, "OperatorState");
+    operatorstate.def(py::init<MWNode<D> &, double*>());
+
+    //OperatorStatistics
+    py::class_<OperatorStatistics<D>> operatorstatistics(m, "OperatorStatistics");
+    operatorstatistics.def(py::init<>())
+        .def("flushNodeCounters", &OperatorStatistics<D>::flushNodeCounters, "flushNodeCounters")
+        .def("incrementFNodeCounters", &OperatorStatistics<D>::incrementFNodeCounters, "incrementFNodeCounters")
+        .def("incrementGNodeCounters", &OperatorStatistics<D>::incrementGNodeCounters, "incrementGNodeCounters");
+
+    //PHOperator
+    py::class_<PHOperator<D>> phoperator(m, "PHOperator");
+    phoperator.def(py::init<MultiResolutionAnalysis<D> &, int>());
+
+    //PoissonOperator
+    py::class_<PoissonOperator, ConvolutionOperator<D>, MWOperator>(m, "PoissonOperator", py::multiple_inheritance())
+        .def(py::init<MultiResolutionAnalysis<D> &, double>(),
+             "MRA"_a,
+             "precision"_a,
+             "ConvolutionOperator: PoissonOperator 1/|r-r'|");
+
+
+    m.def("build_grid",
```

Comment:
Can you move all of the grid functionality into a separate file?

Reply:
yep :)